zap/src/main/java/org/zaproxy/zap/control/AddOnLoader.java

```diff
@@ -126,6 +126,8 @@ public class AddOnLoader extends URLClassLoader {
      */
     private Map<String, AddOnClassLoader> addOnLoaders = new HashMap<>();
 
+    private Map<String, Collection<AddOnClassLoader>> extensionLoaders = new HashMap<>();
```

Comment:
The class loaders are being added but never removed from the map, when the extension/add-on is unloaded.